### PR TITLE
[Overflow-274] Create Page for Team question detail

### PR DESCRIPTION
**Pull request by kentmichaeltelebanco**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-274
## Expected Output
Reused components for question detail page
Can be used to Question Page and Team Question Page
## Notes
Have conditions that would modify the slug inorder to access certain question detail page

## Screenshots
team/[slug]/question/[question_slug]
07 03 2023_17 02 35_REC

questions/[slug]
07 03 2023_00 43 37_REC

### CODE REVIEW

File: backend/app/Console/Kernel.php [hunk 1]
```diff
@@ -10,7 +10,6 @@ class Kernel extends ConsoleKernel
     /**
      * Define the application's command schedule.
      *
-     * @param  \Illuminate\Console\Scheduling\Schedule  $schedule
```
Comment (IanUrriza10) [inline backend/app/Console/Kernel.php]: You might want to undo the changes to these kinds of files since the files changed is increasing more than necessary.

File: frontend/components/organisms/QuestionList/index.tsx [hunk 2]
```diff
@@ -53,12 +74,11 @@ const QuestionList = ({
...
-                        href={`/questions/${slug}`}
-                        className="text-lg text-blue-600 hover:text-blue-400"
-                    >
-                        {title}
-                    </Link>
+                    {page_slug == 'teams'
+                        ? renderTeamQuestionDetailHeader()
+                        : page_slug == 'questions'
+                        ? renderQuestionDetailHeader()
+                        : renderQuestionDetailHeader()}
```
Comment (jules-lucero) [inline frontend/components/organisms/QuestionList/index.tsx]: Can you confirm if both functions are the same? If yes, please refactor to eliminate redundancy.

Maybe try
`page_slug == 'questions' && renderQuestionDetailHeader()`

File: frontend/pages/teams/question/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+export type AnswerEditType = {
+    id: number | null
+    content: string | null
+}
+
+type RefetchType = {
+    slug?: string
+    shouldAddViewCount: boolean
+    answerSort?: { column: string; order: string }[]
+}
```
Comment (jules-lucero) [inline frontend/pages/teams/question/[slug].tsx]: I think no need to redefine these Types, you can just import them since they are exported from the other file

**Review by jules-lucero — DISMISSED** [review]

**Comment by IanUrriza10** [comment]
Review: 1 hr
Comments: 1
Iterations: 3

File: frontend/pages/teams/[slug]/question/[question_slug].tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            answerSort: [{ column: 'VOTES', order: 'DESC' }],
+        },
+        fetchPolicy: 'network-only',
+    })
+
+    if (loading) return loadingScreenShow()
+    else if (error) return errorNotify(`Error! ${error}`)
+    const question: QuestionType = {
+        ...data.question,
+    }
```
Comment (chevyB) [inline frontend/pages/teams/[slug]/question/[question_slug].tsx]:
```suggestion
    const { data: {question}, loading, error, refetch } = useQuery<any, RefetchType>(GET_QUESTION, {
        variables: {
            slug: String(query['question_slug']),
            shouldAddViewCount: true,
            answerSort: [{ column: 'VOTES', order: 'DESC' }],
        },
        fetchPolicy: 'network-only',
    })

    if (loading) return loadingScreenShow()
    else if (error) return errorNotify(`Error! ${error}`)
```
If above is not working, you can use below
`const question: QuestionType = data.question;`

**Comment by ipilikent** [comment]
## Approved PR
- All displayed the same as the Figma design

## Proof
<details>

</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=1238319919
-----
- Total Test time: 1 hrs.- Total Comments: 0
- Total Iterations: 0

**Review by IanUrriza10 — DISMISSED** [review]

**Review by jules-lucero — DISMISSED** [review]

**Comment by ipilikent** [comment]
## Approved PR
- All displayed the same as the Figma design
- The user can click Team Question
- Can redirect Team Question Details
- The slug displayed is correct

## Proof
<details>

274 - Markup Team Details Page

</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=1238319919 
-----
- Total Test time: 1 hr.
- Total Comments: 0
- Total Iterations: 0

**Review by jules-lucero — APPROVED** [review]

**Review by ipilikent — APPROVED** [review]

**Review by IanUrriza10 — APPROVED** [review]
